web/src/stream_popover.js

```diff
@@ -566,9 +570,62 @@ export async function build_move_topic_to_stream_popover(
         event.stopPropagation();
     }
 
+    function caseInsensitiveInlucde(topics, current_topic) {
+        for (const topic of topics) {
+            if (topic.toUpperCase() === current_topic.toUpperCase()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function update_warning_banner() {
+        // The warning banner only shows when
+        // 1) submit button is NOT disabled, and
+        // 2) target topic is an existing topic, and
+        // 3) propagate_mode is NOT change_one
+
+        // submit button
+        if ($("#move_topic_modal .dialog_submit_button")[0].disabled) {
+            $("#move_topic_modal .move_topic_form_warning_container").html("");
+            return;
+        }
+
+        // target topic
+        if (stream_widget_value !== undefined) {
+            const stream_topics = stream_topic_history.get_recent_topic_names(
+                Number.parseInt(stream_widget_value, 10),
+            );
+            const {new_topic_name} = get_params_from_form();
+            if (!caseInsensitiveInlucde(stream_topics, new_topic_name.trim())) {
```

Comment:
This loop structure seems expensive; can't we just modify `PerStreamHistory` in `stream_topic_history` to expose a `has_topic` method, and export that as `stream_topic_history.has_topic(stream_id, topic_name)`? That seems a lot cleaner.